.github/workflows/ci.yml

```diff
@@ -0,0 +1,54 @@
+# SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+#
+# SPDX-License-Identifier: MPL-2.0
+
+name: CI Build
+
+# Controls when the workflow will run.
+on:
+  # run pipeline on push event of main branch.
+  push:
+    branches:
+      - main
+  # run pipeline on pull request
+  pull_request:
+  # run pipeline on merge queue
+  merge_group:
+  # run this workflow manually from the Actions tab
+  workflow_dispatch:
+  schedule:
+    - cron: "0 2 * * *" # Based on UTC time
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-ci-build
+  cancel-in-progress: true
+
+jobs:
+  build-test-release:
+    name: build-test-release
+    uses: "./.github/workflows/build-test-release.yml"
+    permissions:
+      contents: write
+    with:
+      create_release: false
+
+  check-code-quality:
+    uses: "./.github/workflows/check-code-quality.yml"
+
+  reuse-compliance:
+    uses: "./.github/workflows/reuse-compliance.yml"
+
+  clang-tidy:
+    uses: "./.github/workflows/clang-tidy.yml"
+    with:
+      target: "all power_grid_model_benchmark_cpp"
+
+  citations:
+    uses: "./.github/workflows/citations.yml"
```

Comment:
only do this check in nightly or (on PR) when the paths have changed, cfr.
```
  pull_request:
    paths:
      - CITATION.cff
      - .github/workflows/citations.yml
```

Reply:
Because I can't add these lines to ci.yml I will try another method which is:
1. checking out the last commit
2. checking the diff
3. if citation.cff or .github/workflows/citations.yml is in the diff, I run citations